Provide python logging compatibility in eventlet.wsgi

sdague opened the pull request:
Provide a compatibility layer in eventlet.wsgi that allows you
to hand a python logger as the log parameter instead of a
filehandle. This is done via an object wrapper so the logging
methods will only be used if the contents of the log parameter
look like a log, otherwise the previous behavior of just writing
to the logger is done instead.

(This is work in progress patch to make sure the approach is
ok. Can add unit tests in future revision.)

sdague commented:
Apparently this didn't merge last year. This would be hugely helpful in the OpenStack project if it could be. Rebased on current master.

temoto commented:
@sdague hello, sorry for delayed response. Please, take a look at f2aaa0dda commit. Changes: rebased on master again, modified commit subject and (the only important bit) removed "\n" on wsgi.py:843. Please check if removing that `\n` makes sense.

If you care about commit Commiter header, please recommit it. Otherwise I'll merge it as is.

sdague commented:
On 03/30/2015 06:21 PM, Sergey Shepelev wrote:
> @sdague <https://github.com/sdague> hello, sorry for delayed response.
> Please, take a look at f2aaa0d
> <https://github.com/eventlet/eventlet/commit/f2aaa0dda908c1b1a8c75d25d0a781fbfeae7f9f>
> commit. Changes: rebased on master again, modified commit subject and
> (the only important bit) removed "\n" on wsgi.py:843. Please check if
> removing that |\n| makes sense.
> 
> If you care about commit Commiter header, please recommit it. Otherwise
> I'll merge it as is.
> 
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/eventlet/eventlet/pull/75#issuecomment-87857186>.

Looks great, thanks!


-- 
Sean Dague
http://dague.net

temoto commented:
Could you look at test failure https://travis-ci.org/eventlet/eventlet/jobs/56487794 ?

sdague commented:
On 03/31/2015 03:26 AM, Sergey Shepelev wrote:
> Could you look at test failure
> https://travis-ci.org/eventlet/eventlet/jobs/56487794 ?
> 
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/eventlet/eventlet/pull/75#issuecomment-87974088>.
> 

I can't reproduce that locally (tox -e py27-selects). The logging patch
shouldn't be touching anything that's touched by
test_monkeypatched_subprocess at all, so I'm not sure what's going on.

	-Sean

-- 
Sean Dague
http://dague.net

temoto commented:
Yeah, I restarted build and it passed. Must be something temporary in Travis. Merged. Thank you.

temoto commented:
FYI: your patch is included in v0.17.2 release, available on PyPi.
